src/mpid/ch4/shm/posix/posix_rma.h

```diff
@@ -11,233 +11,792 @@
+                                                            void *target_addr,
+                                                            int target_count,
+                                                            MPI_Datatype target_datatype, MPI_Op op)
 {
-    int mpi_errno;
+    int mpi_errno = MPI_SUCCESS;
+    MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_POSIX_COMPUTE_ACCUMULATE);
+    MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_POSIX_COMPUTE_ACCUMULATE);
+
+    /* Handle contig origin datatype */
+    if (MPIR_DATATYPE_IS_PREDEFINED(origin_datatype)) {
+        mpi_errno = MPIDI_CH4U_compute_acc_op(origin_addr, origin_count, origin_datatype,
+                                              target_addr, target_count, target_datatype, op);
+        if (mpi_errno)
+            MPIR_ERR_POP(mpi_errno);
+
+        goto fn_exit;
+    }
+
+    /* Handle derived origin datatype */
+
+    MPI_Datatype basic_type = MPI_DATATYPE_NULL;
+    MPI_Aint predefined_dtp_size = 0, predefined_dtp_extent = 0, predefined_dtp_count = 0;
+    MPI_Aint total_len = 0;
+    MPI_Aint origin_dtp_size = 0;
+    MPIR_Datatype *origin_dtp_ptr = NULL;
+
+    /* Get total length of origin data */
+    MPIR_Datatype_get_size_macro(origin_datatype, origin_dtp_size);
+    total_len = origin_dtp_size * origin_count;
+
+    MPIR_Datatype_get_ptr(origin_datatype, origin_dtp_ptr);
+
+    /* Standard (page 425 in 3.1 report) requires predefined datatype or
+     * a derived datatype where all basic components are of the same predefined datatype.
+     * Thus, basic_type should be correctly set. */
+    MPIR_Assert(origin_dtp_ptr != NULL && origin_dtp_ptr->basic_type != MPI_DATATYPE_NULL);
+
+    basic_type = origin_dtp_ptr->basic_type;
+    MPIR_Datatype_get_size_macro(basic_type, predefined_dtp_size);
+    predefined_dtp_count = total_len / predefined_dtp_size;
+    MPIR_Datatype_get_extent_macro(basic_type, predefined_dtp_extent);
+    MPIR_Assert(predefined_dtp_count > 0 && predefined_dtp_size > 0 && predefined_dtp_extent > 0);
+
+    /* Pack origin data into a contig buffer */
+    MPIR_Segment *seg = NULL;
+    void *packed_buf = NULL;
+    MPI_Aint first = 0, last = total_len;
+
+    packed_buf = MPL_malloc(total_len, MPL_MEM_BUFFER);
+    MPIR_ERR_CHKANDJUMP(packed_buf == NULL, mpi_errno, MPI_ERR_NO_MEM, "**nomem");
+
+    seg = MPIR_Segment_alloc();
+    MPIR_ERR_CHKANDJUMP1(seg == NULL, mpi_errno, MPI_ERR_OTHER, "**nomem", "**nomem %s",
+                         "MPIR_Segment");
+    MPIR_Segment_init(origin_addr, origin_count, origin_datatype, seg);
+    MPIR_Segment_pack(seg, first, &last, packed_buf);
+    MPIR_Segment_free(seg);
+
+    mpi_errno = MPIDI_CH4U_compute_acc_op((void *) packed_buf, (int) predefined_dtp_count,
+                                          basic_type, target_addr, target_count, target_datatype,
+                                          op);
+
+  fn_exit:
+    if (packed_buf)
```

Comment:
Will fix it. Another problem is that, it might be jumped from the place before `packed_buf`'s declaration (e.g., line 39). Will move its declaration into the beginning of the function.